build.gradle

```diff
@@ -112,26 +125,37 @@ spotless {
 check.dependsOn 'spotlessApply'
 
 allprojects {
-    apply plugin: 'base'
+    apply plugin: 'com.bmuschko.docker-remote-api'
 
-    afterEvaluate { project ->
-        def composeDeps = [
-                ":airbyte-config:init",
-                ":airbyte-db:lib",
-                ":airbyte-migration",
-                ":airbyte-scheduler:app",
-                ":airbyte-workers",
-                ":airbyte-server",
-                ":airbyte-webapp",
-        ].toSet().asImmutable()
-
-        if (project.getPath() in composeDeps) {
-            composeBuild.dependsOn(project.getPath() + ':assemble')
-        }
+    task copyDocker(type: Copy) {
+        delete "build/docker"
+
+        from "${project.projectDir}/Dockerfile"
+        into "build/docker/"
+    }
+
+    // Use task types
+    task buildDockerImage(type: DockerBuildImage) {
+        dependsOn copyDocker
+
+        def buildTag = System.getenv('VERSION') ?: 'dev'
+        def buildPlatform = System.getenv('DOCKER_BUILD_PLATFORM') ?: 'linux/amd64'
+        def jdkVersion = System.getenv('JDK_VERSION') ?: '14.0.2'
+        def buildArch = System.getenv('DOCKER_BUILD_ARCH') ?: 'amd64'
+
+       def artifactName = project.getProperties().get("artifactName")
```

Comment:
can we use convention instead of configuration here? and use the name of the project without `airbyte-`? we can still keep the artifactName piece incase someone wants to override it but it could come prepopulated with the correct name so we don't have to put it in the project build fiels.

Reply:
I was doing it, in https://github.com/airbytehq/airbyte/pull/7500/commits/fa899f9a4577184876118e480ce5df7a99b9c623 but there is some corner cases like:
- nested modules like airbyte-scheduler/app, airbyte-config/init, airbyte-db/lib

Reply:
should you check that it is not empty so that it is error proof?